src/renderers/shared/stack/reconciler/ReactRef.js

```diff
@@ -12,12 +12,18 @@
 'use strict';
 
 var ReactOwner = require('ReactOwner');
+var warning = require('warning');
 
 var ReactRef = {};
 
 function attachRef(ref, component, owner) {
   if (typeof ref === 'function') {
-    ref(component.getPublicInstance());
+    var instance = component.getPublicInstance();
+    warning(
+      instance !== null,
+      'Stateless function components cannot have refs.'
```

Comment:
Maybe keep it consistent with the docs and name it "Stateless functional components"?